python_modules/dagster/dagster_tests/daemon_sensor_tests/test_asset_sensor_run.py

```diff
@@ -0,0 +1,660 @@
+            d       f
+            \       /
+                g
+    Sensor for d, f, and g
+    Tests that materializing x, z, and e causes a materialization of d and f, which causes a materialization of g
+    """
+    freeze_datetime = to_timezone(
+        create_pendulum_time(year=2019, month=2, day=27, tz="UTC"),
+        "US/Central",
+    )
+    with instance_with_sensors(attribute="asset_sensor_repo") as (
+        instance,
+        workspace,
+        external_repo,
+    ):
+        with pendulum.test(freeze_datetime):
+            the_sensor = external_repo.get_external_sensor("d_and_f_and_g")
+            instance.start_sensor(the_sensor)
+
+            evaluate_sensors(instance, workspace, executor)
+
+            ticks = instance.get_ticks(the_sensor.get_external_origin_id(), the_sensor.selector_id)
+            assert len(ticks) == 1
+            validate_tick(
+                ticks[0],
+                the_sensor,
+                freeze_datetime,
+                TickStatus.SKIPPED,
+            )
+
+            freeze_datetime = freeze_datetime.add(seconds=60)
+        with pendulum.test(freeze_datetime):
+
+            materialize([x, z, e], instance=instance)
+            wait_for_all_runs_to_finish(instance)
+
+            evaluate_sensors(instance, workspace, executor)
+
+            # sensor should materialize
+            ticks = instance.get_ticks(the_sensor.get_external_origin_id(), the_sensor.selector_id)
+            assert len(ticks) == 2
+            validate_tick(
+                ticks[0],
+                the_sensor,
+                freeze_datetime,
+                TickStatus.SUCCESS,
+            )
+
+            wait_for_all_runs_to_finish(instance)
+            run_request_runs = [r for r in instance.get_runs() if r.pipeline_name == "__ASSET_JOB"]
+            assert len(run_request_runs) == 1
+            assert run_request_runs[0].asset_selection == {
+                AssetKey("d"),
+                AssetKey("f"),
+                AssetKey("g"),
+            }
+
+            freeze_datetime = freeze_datetime.add(seconds=60)
+
+
+@pytest.mark.parametrize("executor", get_sensor_executors())
+def test_lots_of_materializations_sensor(executor):
+    """Asset graph:
+        x
+        |
+        y
+    Sensor for y
+    Tests that materializing x a few times, then starting the sensor results in only one materialization
+        of y
+    """
+    freeze_datetime = to_timezone(
+        create_pendulum_time(year=2019, month=2, day=27, tz="UTC"),
+        "US/Central",
+    )
+    with instance_with_sensors(attribute="asset_sensor_repo") as (
+        instance,
+        workspace,
+        external_repo,
+    ):
+        with pendulum.test(freeze_datetime):
+            for _ in range(5):
+                materialize([x], instance=instance)
+                wait_for_all_runs_to_finish(instance)
+
+            # something is wrong with this test. The in progress asset materialization events are coming in
```

Comment:
very odd... the asset materialization planned events should be created upon run creation